src/SS2ERC721.sol

```diff
@@ -284,20 +273,28 @@ abstract contract SS2ERC721 is ERC721 {
     }
 
     function _burn(uint256 id) internal virtual override {
-        address owner = ownerOf(id);
+        _moveTokenTo(id, BURN_ADDRESS);
+    }
 
-        require(owner != address(0), "NOT_MINTED");
+    function _moveTokenTo(uint256 id, address to) private returns (address owner) {
+        owner = _ownerOfSecondary[id];
 
-        // signed math
-        unchecked {
-            _balanceOfAdjustment[owner]--;
+        if (owner == address(0)) {
+            owner = _ownerOfPrimary(id);
+            require(owner != address(0), "NOT_MINTED");
+
+            _balanceIndicator[owner] |= NOT_PRIMARY_OWNER_FLAG;
```

Comment:
I have a hard time wrapping my head around this: at this point, we know the `owner` is in fact a primary owner, right?

Reply:
I think the approach is sound but the `NOT_PRIMARY_OWNER_FLAG` name is a little confusing.

_balanceIndicator[x] has the flag set when x is a primary owner that transferred their token out, and it means that the actual balance stored in the mapping is the actual running total (this is neat btw 👍)

wdyt of renaming it to something more factual like `SKIP_PRIMARY_BALANCE`?